### ISSUE TITLE
Questions regarding JWT usage for test session access

### ISSUE THREAD

**Issue by abkarcher** [usnistgov/ACVP#1216]
Hello!

We had some questions regarding JWT/accessToken authentication for the sake of requesting validations on test sessions.

We are looking to change our current process so FIPS validations will be requested by people/systems other than the client running the tests. This should nearly eliminate any rejected validation requests submitted due to invalid or mis-formatted metadata. 

This would mean that we need to pass the test session JWT between hands based on the current system, as it is needed to PUT to /testSessions/[tsId]. This did raise some questions however. It would mean two different users had a copy of the JWT - one for submitting a FIPS validation, and the original tester in case they needed to access things about the test session for their personal records or workflow.

We had concerns about these JWTs becoming "out of sync" if one user refreshed the token and the other did not. However, based on some testing I did, it seems  that older versions of the token can also be refreshed without issue, allowing both users to maintain access to test session info. Is this intended, and will  this be the case for the foreseeable future? 

In summary, we are trying to figure out how two separate users can maintain access to a test session without having to constantly trade their access tokens back and forth.

Thanks for your help,
Andrew

**Comment by Kritner** [comment]
Once JWTs are issued, they are valid until their expiration timestamp (unless the client certificate backing the JWT is revoked), so yes you can have multiple JWTs as "currently valid" at any one time.

An potential change that would seemingly accommodate your concern was mentioned here: https://github.com/usnistgov/ACVP-Server/issues/40#issuecomment-732370045. The intention of the comment was to make working on a test session between multiple users within the same company simpler.

Unfortunately, I'm not sure when we will be able to get to that feature at this current time.  Since this issue and https://github.com/usnistgov/ACVP-Server/issues/40  seem to cover more or less the same potential fix, I'll close this for the moment.  If you disagree and/or need further clarification, feel free to reopen/update.

**Comment by Kritner** [comment]
Just as an aside, I don't believe you would need to be *constantly* trade access tokens back and forth - it sounds like you're more or less using a single client cert across several users right?  Since you can have multiple JWTs as "valid" for a single test session through the JWT claims, it would potentially require only communicating the original test session JWT across your multiple users, of whom can refresh their expired JWTs at their own convenience.

This will not address situations such as if multiple users reusing the one certificate attempt to renew the JWT within the same OTP window, but the intended changes from https://github.com/usnistgov/ACVP-Server/issues/40#issuecomment-732370045 will be able to address that.

**Comment by abkarcher** [comment]
Hi,

Thank you for all of the information. We are using a cert across several users, and it sounds like we do not have much an issue here since multiple JWTs can be intentionally valid, and as you said, we can just exchange the original JWT once.

Thanks again!
Andrew
